src/KeyedFrames.jl

```diff
@@ -190,15 +205,17 @@ rename(f::Function, kf::KeyedFrame) = rename!(f, copy(kf))
 
 ##### UNIQUE #####
 
-function Base.unique(kf::KeyedFrame, cols=nothing)
-    return KeyedFrame(unique(frame(kf), cols === nothing ? keys(kf) : cols), keys(kf))
-end
-
-function unique!(kf::KeyedFrame, cols=nothing)
-    unique!(frame(kf), cols === nothing ? keys(kf) : cols)
+_unique(kf::KeyedFrame, cols) = KeyedFrame(unique(frame(kf), cols), keys(kf))
+function _unique!(kf::KeyedFrame, cols)
+    unique!(frame(kf), cols)
     return kf
 end
 
+Base.unique(kf::KeyedFrame, cols::AbstractVector) = _unique(kf, cols)
+Base.unique(kf::KeyedFrame, cols::Union{Integer, Symbol, Colon}) = _unique(kf, cols)
+unique!(kf::KeyedFrame, cols::Union{Integer, Symbol, Colon}) = _unique!(kf, cols)
+unique!(kf::KeyedFrame, cols::AbstractVector) = _unique!(kf, cols)
+
```

Comment:
Yes, because the `Base.unique(kf::KeyedFrame, cols::Union{Integer, Symbol, Colon, AbstractVector})` will throw an ambiguity error. This is primary reason that we need to update KeyedFrames in the first place. Ideally, ambiguities should be resolved from left to right, but that isn't how julia works.

Reply:
```
  MethodError: unique(::KeyedFrame, ::Symbol) is ambiguous. Candidates:
    unique(df::AbstractDataFrame, cols::Union{Colon, Integer, Symbol}) in DataFrames at /Users/rory/.playground/share/keyedframes/depot/packages/DataFrames/KUWId/src/abstractdataframe/abstractdataframe.jl:873
    unique(kf::KeyedFrame, cols::Union{Colon, Integer, Symbol, AbstractArray{T,1} where T}) in KeyedFrames at /Users/rory/repos/KeyedFrames.jl/src/KeyedFrames.jl:215
  Possible fix, define
    unique(::KeyedFrame, ::Union{Colon, Integer, Symbol})
```